parla/parray/core.py

```diff
@@ -0,0 +1,458 @@
+import numpy
+import cupy
+
+from parla.cpu_impl import cpu
+from parla.cuda import gpu
+from parla.tasks import get_current_devices
+from parla.device import Device
+
+class PArray:
+    """Multi-dimensional array on a CPU or CUDA device.
+
+    This class is a wrapper around :class:`numpy.ndarray` and :class:`cupy.ndarray`,
+    It is used to support Parla sheduler optimization and automatic data movement
+
+    Args:
+        array: :class:`cupy.ndarray` or :class:`numpy.array` object
+    """
+    def __init__(self, array=None) -> None:
+        self.array = array
+
+    @property
+    def _on_gpu(self) -> bool:
+        """
+        True if the array is on GPU
+        """
+        return isinstance(self.array, cupy.ndarray)
+
+    @property
+    def _index(self) -> int:
+        """
+        Index of Current Device
```

Comment:
Please update this to:

Return -1 if the current device is CPU.
Otherwise return GPU ID.